modules/netaddress_test.go

```diff
@@ -1,34 +1,91 @@
 package modules
 
 import (
+	"net"
 	"testing"
 )
 
-// TestHost tests the Host method of the NetAddress type.
-func TestHost(t *testing.T) {
+var (
+	// Networks such as 10.0.0.x have been omitted from testing - behavior
+	// for these networks is currently undefined.
+
+	invalidAddrs = []string{
+		// Garbage addresses
+		"",
+		"foo:bar:baz",
+		"garbage:6146:616",
+		// Missing host / port
+		":",
+		"111.111.111.111",
+		"12.34.45.64",
+		"[::2]",
+		"::2",
+		"foo",
+		"hn.com",
+		"世界",
+		"foo:",
+		"世界:",
+		":foo",
+		":世界",
+		// Invalid host / port chars
+		" foo:bar",
+		"foo :bar",
+		"f oo:bar",
+		"foo: bar",
+		"foo:bar ",
+		"foo:b ar",
+		"\x00:bar",
+		"foo:\x00",
+		// Unspecified address
+		"[::]:bar",
+		"0.0.0.0:bar",
+	}
+	validAddrs = []string{
+		// Loopback address (valid in testing only, can't really test this well)
+		"localhost:bar",
+		"127.0.0.1:bar",
+		"[::1]:bar",
+		// Valid addresses.
+		"foo:bar",
+		"hn.com:8811",
+		"[::2]:bar",
+		"111.111.111.111:111",
+		"12.34.45.64:7777",
+		"世界:bar",
+		"bar:世界",
+		"世:界",
+		`":"`, // yeah, okay
```

Comment:
ಠ_ಠ

seems easier to me to call `strconv.Atoi` on the port.
```go
portInt, err := strconv.Atoi(port)
if err != nil || portInt < 0 || portInt > 65535 {
    return errors.New("invalid port")
}
```
this would catch unicode stuff too